packages/babel-parser/src/parser/expression.js

```diff
@@ -893,6 +893,43 @@ export default class ExpressionParser extends LValParser {
     return node;
   }
 
+  parseClassOrClassHasInstanceExpression<T: N.Class>(
+    node: T,
+    isStatement: boolean,
+  ): N.Class | N.ClassHasInstanceExpression {
+    this.next();
+    if (this.eat(tt.dot)) {
+      if (this.isContextual(tt._hasInstance)) {
+        this.expectPlugin("classBrandCheck");
+        return this.parseClassHasInstanceExpression(
+          node,
+          this.startPos,
+          this.startLoc,
+        );
+      } else {
+        throw this.raise(Errors.UnexpectedToken, { at: node });
+      }
+    }
+    this.takeDecorators(node);
+    return this.parseClass(node, isStatement);
+  }
+
+  parseClassHasInstanceExpression(base, startPos, startLoc) {
+    this.next(); // eat `hasinstance`
+    this.next(); // eat `(`
+    const node = this.startNodeAt(startPos, startLoc);
+    const argus = this.parseCallExpressionArguments(tt.parenR);
+    if (argus.length !== 1) {
+      this.raise(Errors.InvalidArguments, { at: startLoc });
+    }
+
+    node.instance = argus[0];
+
+    this.state.stop = true;
```

Comment:
Why do we define a `stop` property here?

Reply:
Unused code. It was my mistake.